template-processors/openshift-provision/files/processParameters.yml

```diff
@@ -0,0 +1,41 @@
+---
+- name: Process parameters
+  hosts: localhost
+  gather_facts: false
+  tasks:
+  - set_fact:
+      _cluster_definition_dir: "{{ openshift_cluster_config }}/cluster/{{ cluster_name }}"
+
+  # Build the hierarchy for your configuration defined for a given cluster
+  - include_vars:
+      file: '{{ _cluster_definition_dir }}/vars/main.yml'
+
+  - name: Initialize hierarchy_paths
+    set_fact:
+      hierarchy_paths: ["{{ openshift_cluster_config }}/default"]
+
+  - name: Define dynamic hierarchy_paths
+    set_fact:
+      hierarchy_paths: "{{ hierarchy_paths }} + [ '{{ openshift_cluster_config }}/{{ item.key }}/{{ item.value }}' ]"
+    with_dict: "{{ hierarchy }}"
+
+  - name: Include hierarchy in top level vars
+    set_fact:
+      "{{ item.key }}": "{{ item.value }}"
+    with_dict: "{{ hierarchy }}"
+
+  - name: Include vars from each level of hierarchy
+    include_vars:
+      dir: "{{ item }}/vars"
+      files_matching: '^[a-zA-Z0-9].*\.(ya?ml|json)$'
```

Comment:
I grabbed that from one of the examples listed in the openshift-provision README. 
https://github.com/gnuthought/openshift-provision-manager/blob/master/runner/openshift-provision.yaml#L30

I was torn I wanted to include `json` in the retrieving of the `vars` or not, because Ansible will allow you to import json files as variables, but Kubernetes and yaml...I'd be open to changing it if people have a preference on it.

Reply:
Basically, I'm ok with whichever way you choose, just please make sure it's documented in the readme — both (1) and (2).

By the way: personally, whenever a reviewer asks a question, I taught myself to interpret it as a fairly unambigious, if indirect, signal that the answer is not clear enough from the existing code and documentation. As such, I like to try and actually *avoid* answering this in a review reply — instead, change the documentation itself (and/or add an inline comment in the code), and ask the reviewer if they find it clear enough from the code after the change! :) I believe this has a nice benefit of hopefully making it easier to understand for any future readers too, whereas review replies are much more ephemeral — while the amount of actual typing is mostly the same in both cases.